src/utils/proxyMap.ts

```diff
@@ -0,0 +1,114 @@
+import { proxy } from 'valtio'
+
+type InternalProxyMap<K, V> = Map<K, V> & {
+  data: {
+    [key: string]: V
+  }
+  toJSON: object
+  makeMap(): Map<K, V>
+}
+
+export const proxyMap = <K, V>(entries?: [K, V][] | null): Map<K, V> => {
```

Comment:
```suggestion
export const proxyMap = <K, V>(entries?: Iterable<readonly [K, V]> | null): Map<K, V> => {
```

We want to support `proxyMap(new Map(...))` too.